glib/src/param_spec.rs

```diff
@@ -377,96 +377,66 @@ macro_rules! define_param_spec_min_max {
 }
 
 macro_rules! define_param_spec_numeric {
-    ($rust_type:ident, $ffi_type:path, $value_type:ty, $rust_type_offset:expr, $from_glib:expr) => {
+    ($rust_type:ident, $ffi_type:path, $value_type:ty, $rust_type_offset:expr, $ffi_fun:ident, $alias:literal) => {
         define_param_spec!($rust_type, $ffi_type, $rust_type_offset);
-        define_param_spec_default!($rust_type, $value_type, $from_glib);
-        define_param_spec_min_max!($rust_type, $value_type, $from_glib);
+        define_param_spec_default!($rust_type, $value_type, |x| x);
+        define_param_spec_min_max!($rust_type, $value_type, |x| x);
+
+        impl $rust_type {
+            #[allow(clippy::new_ret_no_self)]
+            #[doc(alias = $alias)]
+            pub fn new(
+                name: &str,
+                nick: &str,
+                blurb: &str,
+                minimum: $value_type,
+                maximum: $value_type,
+                default_value: $value_type,
+                flags: ParamFlags,
+            ) -> ParamSpec {
+                assert_param_name(name);
+                unsafe {
+                    from_glib_none(gobject_ffi::$ffi_fun(
+                        name.to_glib_none().0,
+                        nick.to_glib_none().0,
+                        blurb.to_glib_none().0,
+                        minimum,
+                        maximum,
+                        default_value,
+                        flags.into_glib(),
+                    ))
+                }
+            }
+        }
     };
 }
 
+fn assert_param_name(name: &str) {
+    assert!(
+        is_canonical_pspec_name(name),
+        "{} is not a valid canonical parameter name",
+        name
+    );
+}
 define_param_spec_numeric!(
     ParamSpecChar,
     gobject_ffi::GParamSpecChar,
     i8,
     0,
-    |x| x
+    g_param_spec_char,
+    "g_param_spec_char"
```

Comment:
You should be able to use [`stringify!`](https://doc.rust-lang.org/std/macro.stringify.html) for that

Reply:
I wanted to use `stringify!` as well, but the macro call ends up here
`#[doc(alias = stringify!($ffi_fun))]`
and macro calls in that position are not possible, because `alias` expects a literal.

Reply:
What if you make `alias` expect a `expr` instead?

Reply:
Oh, you can't. You mean the doc alias needs that. Can you create an issue about that for rustc? Seems fine like this here then

Reply:
>  Can you create an issue about that for rustc?

https://github.com/rust-lang/rust/issues/94180